app/src/main/java/org/mozilla/focus/observer/LoadingObserver.java

```diff
@@ -0,0 +1,46 @@
+package org.mozilla.focus.observer;
+
+import android.arch.lifecycle.LifecycleOwner;
+import android.os.SystemClock;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import org.mozilla.focus.architecture.NonNullObserver;
+import org.mozilla.focus.session.Session;
+import org.mozilla.focus.telemetry.TelemetryWrapper;
+
+public class LoadingObserver {
+
+    private static final String LOG_TAG = "LoadingObserver";
+
+    public LoadingObserver(LifecycleOwner owner, final Session session) {
+        session.getLoading().observe(owner, new NonNullObserver<Boolean>() {
+
+            private long startLoadTime = 0;
+            private boolean loadStart = false;
+
+            @Override
+            public void onValueChanged(@NonNull Boolean loading) {
+                if (loading) {
+                    if (!loadStart) {
+                        startLoadTime = SystemClock.elapsedRealtime();
+                        Log.i(LOG_TAG, "zerdatime " + startLoadTime +
+                                " - page load start");
+                        TelemetryWrapper.startLoad(startLoadTime);
+                        loadStart = true;
+                    }
+                } else {
+                    if (loadStart) {
+                        Log.i(LOG_TAG, "Loaded page at " + session.getUrl().getValue());
+                        long endTime = SystemClock.elapsedRealtime();
+                        Log.i(LOG_TAG, "zerdatime " + endTime +
+                                " - page load stop");
+                        Log.i(LOG_TAG, (endTime - startLoadTime) + " - elapsed load");
+                        TelemetryWrapper.endLoad(endTime);
+                        loadStart = false;
+                    }
```

Comment:
Somehow we are duplicating the logic here. The same checks (loadStart/started) and time calculations happen in TelemetryWrapper too. I wonder if we can clean this up a little. Maybe this class here does the time calculation on its own and just calls something like reportLoadTime() on TelemetryWrapper. After all TelemetryWrapper only needs the load time to calculate a new average.